### PR TITLE
Add project specific permissions

### PR DESCRIPTION
**Pull request by singhabahu**
join table `users_projects` now has a new `permission` field to manage project specific permissions. We can include permissions as a JSON string. This will follow the same pattern as the organization level permissions for the consistency.

### CODE REVIEW

File: src/db/models/project.js [hunk 1]
```diff
@@ -28,5 +28,9 @@ export const Project = sequelize.define('projects', {
         tableName: 'projects',
     });
 
-Project.belongsToMany(User, {through: 'users_projects'});
-User.belongsToMany(Project, {through: 'users_projects', foreignKey: 'userId'});
+const UserProjects = sequelize.define('users_projects', {
+    permission: Sequelize.TEXT('long'),
```
Comment (Rishanthakumar) [inline src/db/models/project.js]: @singhabahu can't we use type: Sequelize.JSON ? like what we did for role permission. If you are going to use just an array this is okay.

File: src/util/permission.js [hunk 2]
```diff
@@ -9,27 +9,29 @@ import {Role} from './../db/models/role';
...
-        }).catch((error) => {
-            reject(error);
-        });
+  return new Promise((resolve, reject) => {
+    User.find({
+      attributes: ['uuid'],
+      where: {
+        uuid: userId,
+      },
+      include: [{
```
Comment (Rishanthakumar) [inline src/util/permission.js]: @singhabahu don't we need to get the userProject joining table permission here ?

**Review by Rishanthakumar — CHANGES_REQUESTED** [review]

**Comment by singhabahu** [comment]
@Rishanthakumar I tried to change the TEXT feild to JSON in the DB model but it adds lot of code overhead. We'll do that later. Let's keep this simple for now.

**Review by Rishanthakumar — APPROVED** [review]
Looks good.

**Pull request merged**